opendistro-for-elasticsearch/plan.sh

```diff
@@ -0,0 +1,94 @@
+ELASTICSEARCH_VERSION="6.5.4"
+ELASTICSEARCH_PKG_URL="https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-$ELASTICSEARCH_VERSION.tar.gz"
+ELASTICSEARCH_PKG_SHASUM="0943f434170567e6d780d24caca86d23710049599172a533f8140b2f659dd130"
+pkg_version=0.7.0.1
+pkg_name="opendistro-for-elasticsearch"
+pkg_description="An Apache 2.0-licensed distribution of Elasticsearch enhanced with enterprise security, alerting, SQL, and more"
+pkg_origin="core"
+pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
+pkg_license=("Apache-2.0")
+pkg_upstream_url="https://github.com/opendistro-for-elasticsearch"
+pkg_build_deps=(
+  core/git
+  core/maven
+  core/zip
+)
+pkg_deps=(
+  core/coreutils
+  core/curl
+  core/glibc
+  core/openjdk11
+  core/openssl
+  core/procps-ng
+  core/ruby
+  core/zlib
+)
+
+pkg_bin_dirs=(bin)
+pkg_binds_optional=(
+  [elasticsearch]="http-port transport-port"
+)
+pkg_lib_dirs=(lib)
+pkg_exports=(
+  [http-port]=es_yaml.http.port
+  [transport-port]=es_yaml.transport.tcp.port
+)
+pkg_exposes=(http-port transport-port)
+
+OPENDISTRO_DEPENDENCIES=(
+  'security-parent'
+  'security-ssl'
+  'security-advanced-modules'
+)
+
+do_download() {
+  download_file "${ELASTICSEARCH_PKG_URL}" "${HAB_CACHE_SRC_PATH}/elasticsearch-oss-${ELASTICSEARCH_VERSION}.tar.gz" "${ELASTICSEARCH_PKG_SHASUM}"
+  for component in "${OPENDISTRO_DEPENDENCIES[@]}"; do
+    rm -rf "${HAB_CACHE_SRC_PATH:?}/${component}"
+    git clone "https://github.com/opendistro-for-elasticsearch/${component}.git" "${HAB_CACHE_SRC_PATH}/${component}"
```

Comment:
You could potentially save on download speed and time by changing the clone line to get the branch you require, and limiting depth.

A great example of this can be found in the [hugo](https://github.com/habitat-sh/core-plans/blob/master/hugo/plan.sh#L21) plan.

This eliminates the need for a `checkout` later in the plan file.

Reply:
Wow this is such a huge time save it's insane. This cut my build times from ~7m to ~3m. I think it's all of the diskio that pulling the entire commit tree incurs (the download speed is actually fine). When we pull directly from the branch with the `depth` operator, we only get the commits from just that one branch/tag.